all-actors/src/main/java/org/sunbird/notification/dispatcher/impl/FCMNotificationDispatcher.java

```diff
@@ -79,24 +84,77 @@
 
   /** Initialises Kafka producer required for dispatching messages on Kafka. */
   private void initKafkaClient() {
-    Config config = ConfigUtil.getConfig();
-    BOOTSTRAP_SERVERS = config.getString(Constant.SUNBIRD_NOTIFICATION_KAFKA_SERVICE_CONFIG);
-    topic = config.getString(Constant.SUNBIRD_NOTIFICATION_KAFKA_TOPIC);
+    if (producer == null) {
+      Config config = ConfigUtil.getConfig();
+      BOOTSTRAP_SERVERS = config.getString(Constant.SUNBIRD_NOTIFICATION_KAFKA_SERVICE_CONFIG);
+      topic = config.getString(Constant.SUNBIRD_NOTIFICATION_KAFKA_TOPIC);
 
-    logger.info(
-        "KafkaTelemetryDispatcherActor:initKafkaClient: Bootstrap servers = " + BOOTSTRAP_SERVERS);
-    logger.info("UserMergeActor:initKafkaClient: topic = " + topic);
-    try {
-      producer =
-          KafkaClient.createProducer(
-              BOOTSTRAP_SERVERS, Constant.KAFKA_CLIENT_NOTIFICATION_PRODUCER);
-    } catch (Exception e) {
-      logger.error("UserMergeActor:initKafkaClient: An exception occurred.", e);
+      logger.info(
+          "KafkaTelemetryDispatcherActor:initKafkaClient: Bootstrap servers = "
+              + BOOTSTRAP_SERVERS);
+      logger.info("UserMergeActor:initKafkaClient: topic = " + topic);
+      try {
+        producer =
+            KafkaClient.createProducer(
+                BOOTSTRAP_SERVERS, Constant.KAFKA_CLIENT_NOTIFICATION_PRODUCER);
+      } catch (Exception e) {
+        logger.error("UserMergeActor:initKafkaClient: An exception occurred.", e);
+      }
     }
   }
 
   @Override
   public boolean dispatchAsync(Map<String, Object> data) {
-    return false;
+    boolean response = true;
+    initKafkaClient();
+    List<Map<String, Object>> notificationDataList =
+        (List<Map<String, Object>>) data.get(NOTIFICATIONS);
+    for (int i = 0; i < notificationDataList.size(); i++) {
+      Map<String, Object> innerMap = (Map<String, Object>) notificationDataList.get(i);
+      String message = processTopicData(innerMap);
+      ProducerRecord<Long, String> record = new ProducerRecord<>(topic, message);
+      if (producer != null) {
+        producer.send(record);
+      } else {
+        response = false;
+        logger.info("UserMergeActor:mergeCertCourseDetails: Kafka producer is not initialised.");
+      }
+    }
+    return response;
+  }
+
+  private String processTopicData(Map<String, Object> data) {
+    KafkaMessage message = new KafkaMessage();
```

Comment:
I would have had a overloaded constructor or by default populated some members to avoid writing so big a function.
Also, this seems best to name as getTopicMessage()

Reply:
@indrajra chnages is done.